# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/shadow-10-uclibc-innetgr.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- shadow-4.0.6/libmisc/login_access.c.orig	2003-05-05 23:44:15.000000000 +0200
+++ shadow-4.0.6/libmisc/login_access.c	2004-11-23 09:21:22.672325008 +0100
@@ -192,6 +192,7 @@
 static int
 netgroup_match (const char *group, const char *machine, const char *user)
 {
+#ifdef NIS
 	static char *mydomain = 0;
 
 	if (mydomain == 0) {
@@ -202,6 +203,10 @@
 	}
 
 	return innetgr (group, machine, user, mydomain);
+#else
+	syslog(LOG_ERR, "NIS netgroup support not configured");              
+	return (NO);
+#endif
 }
 
 /* user_match - match a username against one token */
